.dockerignore

```diff
@@ -0,0 +1,32 @@
+# ignore docker config
+docker-compose.yml
+Dockerfile
+.dockerignore
+nginx.conf
+
+# ignore repo docs
+EDITING.md
+README.md
+LICENSE.txt
+.git*
+
+# ignore secrets / CI logic
+build_key.enc
+push-build.sh
+.travis.yml
+
+# ignore unnecessary certbot files
+_docs
+!_docs/tools
+!_docs/acme
+!_docs/certbot
+!_docs/certbot-*
+!_docs/letsencrypt-auto-source
+
+# ignore build artifacts
+package-lock.json
```

Comment:
Please forgive my node ignorance, but if we don't ignore this file, won't we reuse the versions of dependencies pinned in `package-lock.json` in the Docker image? At least at face value and my experience with dependencies in Python, this seems valuable since it keeps the packages pinned to known working versions and reuses the same versions as I believe are used in Travis.

Reply:
That's right. I agree that it wasn't great to ignore the lock file, I just wasn't able to get `npm install` working with package-lock.json on recent versions of node. I just locked the container's node version to 8.12 to match what's in .travis.yml, and that seems to have fixed the issues. Thanks for flagging this!